src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,40 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
+Animal.remove = function() {
+  Animal.alive = Animal.alive.filter((obj) => obj.health !== 0);
+};
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (!animal.hidden && (animal instanceof Carnivore) === false) {
```

Comment:
you could check for Herbivore
```suggestion
    if (!animal.hidden && animal instanceof Herbivore) {
```